sphinx/writers/latex.py

```diff
@@ -462,12 +465,17 @@ def declare_package(packagename, options=None):
             if tocdepth >= SECNUMDEPTH:
                 # Increase secnumdepth if tocdepth is depther than default SECNUMDEPTH
                 self.elements['secnumdepth'] = '\\setcounter{secnumdepth}{%d}' % tocdepth
-        if getattr(document.settings, 'contentsname', None):
-            self.elements['contentsname'] = \
-                self.babel_renewcommand('\\contentsname', document.settings.contentsname)
         # allow the user to override them all
         self.check_latex_elements()
         self.elements.update(builder.config.latex_elements)
+
+        if self.elements['babel'] != '3.1415':
+            self.elements['multilingual'] = self.elements['babel']
```

Comment:
I think I am a bit tired here and can not think straight anymore... I think this trick is fine. Use of ``'babel'`` by user must overrule latex writer default choices, independently of ``latex_engine``.